syntax = "proto3";

package remote;

option java_package = "com.lsorter.detection.detectors";
option java_outer_classname = "LegoBrickProto";

message ImageRequest {
  bytes image = 1;
  int32 rotation = 2;
}

message ImageStore {
  bytes image = 1;
  int32 rotation = 2;
  string label = 3;
}

message ListOfBoundingBoxes {
  repeated BoundingBox packet = 1;
}

message BoundingBox {
  int32 ymin = 1;
  int32 xmin = 2;
  int32 ymax = 3;
  int32 xmax = 4;
  string label = 5;
  float score = 6;
}

message Empty {

}

service LegoBrick {
  rpc CollectImages (ImageStore) returns (Empty);
  rpc CollectCroppedImages (ImageStore) returns (Empty);
  rpc DetectBricks (ImageRequest) returns (ListOfBoundingBoxes);
  rpc DetectAndClassifyBricks (ImageRequest) returns (ListOfBoundingBoxes);
}